projects/storefrontapp-e2e-cypress/cypress/e2e/regression/my-account/my-account-v2-orders.e2e.cy.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { fillLoginForm } from '../../../helpers/auth-forms';
+import { viewportContext } from '../../../helpers/viewport-context';
+import { isolateTests } from '../../../support/utils/test-isolation';
+
+describe(
+  'My Account Version-2 Order History and Order Details',
+  { testIsolation: false },
+  () => {
+    viewportContext(['desktop'], () => {
+      isolateTests();
+      before(() => {
+        cy.window().then((win) => win.sessionStorage.clear());
+        cy.visit('/');
+      });
+
+      beforeEach(() => {
+        cy.restoreLocalStorage();
+      });
+
+      it('should navigate to login page and SignIn with user details', () => {
+        cy.findByText(/Sign in \/ Register/i).click();
+        fillLoginForm({ username: 'cdp.user@sap.com', password: 'Test@1' });
+      });
+
+      it('should navigate to Order History page', () => {
+        let totalCount;
+        cy.findByText(/My Account/i).click();
+        cy.findByText(/Order History/i).click();
+        cy.get('h2').contains('All Orders');
+        cy.get('.cx-myaccount-v2-order-history-body > .cx-each-order').then(
+          (value) => {
+            totalCount = Cypress.$(value).length;
+            expect(value).to.have.length(totalCount);
+          }
```

Comment:
[QUESTION]
Could you help me understand in which conditions this test will NOT pass?
Maybe I'm missing something obvious, but at the first sight it looks to me that it will always pass, regardless of the conditions.

Reply:
I have replaced the value check now as below
expect(totalCount).to.be.at.least(1); 

It will NOT PASS when there is no order.